src/renderers/shared/utils/__tests__/ReactErrorUtils-test.js

```diff
@@ -71,5 +72,15 @@ describe('ReactErrorUtils', () => {
       __DEV__ = true;
       global.process = oldProcess;
     });
+
+    it('shouldn\'t call the callback again while executing synchronously induced other call in development', () => {
```

Comment:
Is there any way to write a test for observable behavior instead? That is, write a test that relies on `ReactDOM` and `React`, but not on `ReactErrorUtils` itself. Otherwise, it's hard to tell what exactly this is fixing.